.github/workflows/build.yml

```
@@ -98,3 +98,55 @@ jobs:
       - name: Setup tmate debug session
         uses: mxschmitt/action-tmate@v3
         if: env.RUN_TMATE
+  test:
+    name: test source - go${{ matrix.go-version }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-latest
+        go-version:
+          - "1.19"
+          - "1.20"
+        include:
+          - os: ubuntu-20.04
+            go-version: "1.20"
+    steps:
+      - uses: actions/checkout@v3
+      - id: setup-go
+        uses: actions/setup-go@v4
+        with:
+          go-version: ${{ matrix.go-version }}
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-golang-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-golang-
+
+      - name: Retrieve package location
+        run: |
+            echo PACKAGE_LOCATION="$(find ./ -name 'test')/../" >> $GITHUB_OUTPUT
+        id: package-location
+
+      - name: Generate mocks and run tests
+        working-directory: ${{ steps.package-location.outputs.PACKAGE_LOCATION }}
+        run: |
+          export GOPATH="$HOME/go/"
+          export PATH=$PATH:$GOPATH/bin
+          go install github.com/golang/mock/mockgen@latest
+          go install github.com/jstemmer/go-junit-report@latest
+          export PACKAGE_LOCATION="$(find ./ -name 'test')/../"
+          go generate ./
+          go test -v ./test/* | go-junit-report -set-exit-code > report.xml
+
+      - name: Test summary
+        uses: test-summary/action@v1
+        with:
+          paths: |
+            ${{ steps.package-location.outputs.PACKAGE_LOCATION }}/report.xml
+        if: always()  # If previous steps fail we still want to \
+                      # update build status since it will be affected
```

Minor style thing- for a comment that will put us over the 88 character line limit, we typically put it on the previous line(s):
```suggestion
        # If previous steps fail, we still want to always update build status
        # since it will be affected.
        if: always()
```